dev/ledger.md

```diff
@@ -751,6 +857,50 @@ An asset freeze transaction additionally has the following fields:
 The cryptographic hash of the fields above is called the _transaction
 identifier_.  This is written as $\Hash(\Tx)$.
 
+## State proof transaction
+
+A special transaction is used to disseminate and store state 
+proofs.  The type of a state proof transaction is `stpf`.
+This type of transaction must always be issued from a special sender
+address, which is the hash of the domain-separation prefix `SpecialAddr`
+with the string `StateProofSender`.  The transaction must not have any
+signature, must not have any fee, must have an empty note, must not have
+the rekeying field set, must not have any lease, and must not be part
+of a transaction group. 
+State proof transaction verification does not apply any special constrains on the first and last valid parameter.
+The state proof transaction includes four additional fields:
+
+ - Under msgpack key `sptype`, the type of the state proof; currently always zero.
+ - Under msgpack key `sprnd`, the last round that this state proof attest to.
+ - Under msgpack key `sp`, the state proof fields as defined in the state proof format subsection 
+   of the cryptographic primitive specification document.
+ - Under msgpack key `spmsg`, a structure that compose the state proof message, whose hash is being attested
+   by the state proof. This structure defined [above](#state-proof-message)
+
+In order for a state proof transaction to be valid, the round of
+the state proof (`sprnd`) must be exactly equal to the next
+expected state proof round in the block header, as described
+[above](#state-proof-tracking).  When a state proof
+transaction is applied to the state, the next expected state proof round
+for that type of state proof is incremented by $\delta_{SP}$.
+To encourage the formation of shorter state proof, the rule for
+validity of state proof transactions is dependent on the first valid `fv` round
+number in the transaction.
+In particular, the signed weight of a state proof must be:
+
+- Equal to the total online stake, `TotalWeight`, if the
```

Comment:
That seems wrong. The largest that the signed weight can be is the total weight of the top $N_{SP}$ accounts, not the total online weight.

If I understand it correctly, you have here two quantities: `Top` = total-weight-of-top-$N_{SP}$-accounts, and `ProvenWeight` = (total-online-stake * $f_{SP}$) / 2^32. 

The "signed weight" (which really needs to get a name) starts from `Top` when `fv` is $\delta_{SP}/2$ since the last proof (or less). Once `fv` hits that round number, the "signed weight" is decreased linearly as `fv` is increased. When `fv` hits $\delta_{SP}$ since the last proof, the signed weight is equal to `ProvenWeight`. After that, the signed weight does not decrease anymore.

Setting `Top` = total-online-stake (as is done in the text above) is just wasting the rounds from $\delta_{SP}/2$ until the signed weight goes below total-weight-of-top-$N_{SP}$-accounts. (Since the total signed weight can never be more than that.)